File: ruby/gameplay_db_adapter/lib/gameplay/in_memory_board_repository.rb
require 'gameplay/board_repository'

module Gameplay
  class InMemoryBoardRepository
    include BoardRepository

    def initialize
      @store = {}
    end

    def save(board)
      @store[board.id] = board
    end

    def find_by_id(board_id)
      @store[board_id]
    end
  end
end

File: java/gameplay-db-adapter/src/main/java/com/hexagonalchess/gameplay/InMemoryBoardRepository.java
package com.hexagonalchess.gameplay;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class InMemoryBoardRepository implements BoardRepository {

    private final Map<BoardId, Board> store = new HashMap<>();

    @Override
    public void save(Board board) {
        store.put(board.id(), board);
    }

    @Override
    public Optional<Board> findById(BoardId boardId) {
        return Optional.ofNullable(store.get(boardId));
    }
}
